### PR TITLE
Refactor fault/error logic

### PR DESCRIPTION
**Pull request by harrryr**
**Description:** 
This PR just refactors the logic in the awsxrayexporter as it was hard to follow the logical flow of the code. No functional changes.

**Link to tracking Issue:** 

**Testing:**
Ran unit tests
Ran integration test by creating eks cluster and activating sample app.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/awsxrayexporter/internal/translator/cause.go [hunk 1]
```diff
@@ -118,41 +118,34 @@ func makeCause(span ptrace.Span, attributes map[string]pcommon.Value, resource p
 		}
 	}
 
+	// The logic to record error and fault should be kept in sync with the ADOT pulse repo whenever possible except for the throttle
+    // https://github.com/aws-observability/aws-apm-java-instrumentation/blob/main/awsagentprovider/src/main/java/software/amazon/opentelemetry/javaagent/providers/AwsSpanMetricsProcessor.java
```
Comment (songy23) [inline exporter/awsxrayexporter/internal/translator/cause.go:122]: nit: format seems off

**Review by songy23 — APPROVED** [review]
LGTM, pinging code owners for review @wangzlei, @srprash

File: exporter/awsxrayexporter/internal/translator/cause.go [hunk 2]
```diff
@@ -118,41 +118,35 @@ func makeCause(span ptrace.Span, attributes map[string]pcommon.Value, resource p
...
-			isError = false
-			isThrottle = false
-			isFault = true
-		case status.Code() == ptrace.StatusCodeError:
+	var code int64
+	if ok {
+		code = val.Int()
+	}
+
+	isThrottle = false
```
Comment (pxaws) [inline exporter/awsxrayexporter/internal/translator/cause.go:132]: Can we have a default false value for isError and isFault as well? This can simplify the switch block.

**Review by pxaws — APPROVED** [review]

**Review by wangzlei — APPROVED** [review]
LGTM

**Review by dmitryax — APPROVED** [review]
